js/form.js

```diff
@@ -269,12 +269,20 @@ function setupBirthday(limits) {
 function isValid(element, value) {
   switch (element) {
     case "nickname":
+      let numberSpecialCharacters = 0
+      let nameWithoutSpecialCharacters = ''
+
+      for (let i = 0; i < value.length; i++) {
+        if (/[^a-zA-Z0-9\u00c0-\u017e\u0400-\u04FF ]/.test(value[i])) numberSpecialCharacters++
+        else nameWithoutSpecialCharacters += value[i]
+      }
+
       return (
         value.length >= 1 &&
         value.length <= 80 &&
-        /^([A-Za-z0-9 ]*[A-Za-z0-9][A-Za-z0-9 ]*[^A-Za-z0-9 ]?[A-Za-z0-9 ]*[^A-Za-z0-9 ]?[A-Za-z0-9 ]*|[A-Za-z0-9 ]*[^A-Za-z0-9 ]?[A-Za-z0-9 ]*[A-Za-z0-9][A-Za-z0-9 ]*[^A-Za-z0-9 ]?[A-Za-z0-9 ]*|[A-Za-z0-9 ]*[^A-Za-z0-9 ]?[A-Za-z0-9 ]*[^A-Za-z0-9 ]?[A-Za-z0-9 ]*[A-Za-z0-9][A-Za-z0-9 ]*)$/.test(
-          value
-        )
+        value.trim() === value &&
+        numberSpecialCharacters <= 2 &&
+        /^[a-zA-Z0-9\u00c0-\u017e\u0400-\u04FF ]+$/.test(nameWithoutSpecialCharacters)
```

Comment:
I'm also not sure what you're testing here. Since you removed the special characters earlier with the inverse RegExp it seems logical that this would always be true. Am I missing something?

Reply:
Should we centralize the validation function into a single js file? It's kind of ugly to have it in two places in the code imho. I'd be fine with adding an extra .js file just for validations that is loaded in both the form and the submit page.

Reply:
It tests to ensure that there are no Leading or Trailing whitespace characters, and that we have two or fewer special characters, as required by the server. We then test to ensure that remaining characters include at least one "regular" character.

I think moving validation to a single js file would be the best scenario. I'll include that in this PR. :)